public/app/features/provisioning/Wizard/WizardContent.tsx

```diff
@@ -94,8 +94,11 @@ export function WizardContent({
   const handleRepositoryDeletion = async (name: string) => {
     try {
       await deleteRepository({ name });
-      // Wait before redirecting to ensure deletion is indexed
-      setTimeout(() => navigate(PROVISIONING_URL), 1500);
+      // Wait before redirecting to ensure deletion is processed
+      setTimeout(() => {
+        settingsQuery.refetch();
+        navigate(PROVISIONING_URL);
+      }, 1500);
```

Comment:
This doesn't really work, once `deleteRepository` returns success, RTK refetches the repos, and if there's none, it redirects automatically to the starting page (because there's nothing to render here), so this code isn't even called. It seems like the settings are not updated at the same time.